test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -615,6 +618,116 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
+			}
+			runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			p.Spec.RuntimeClassName = &runtimeClass
+			p.Spec.Containers[0].Image = images.Test()
+			p.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("200Mi"),
+					corev1.ResourceCPU:    resource.MustParse("2"),
+				},
+			}
+
+			// Adding a unique label to the deployment
+			deploymentLabels := map[string]string{
+				"app": DeploymentName,
+			}
+			// we delete the deployment either way
+			defer func() {
+				err := testclient.DataPlaneClient.Get(ctx, client.ObjectKey{Name: dp.Name, Namespace: testutils.NamespaceTesting}, dp)
+				if err == nil {
+					// deployment exists, delete it
+					testlog.Infof("Deleting Deployment %v", dp.Name)
+					err := testclient.Client.Delete(ctx, dp)
+					Expect(err).ToNot(HaveOccurred())
+				}
+				testlog.Infof("Deployment %v is deleted %v", dp.Name)
+			}()
+			// we create and delete deployment in loop to create deployments in
+			// quick succession to verify pre-start hook is able to write to
+			// cpuset.cpus.exclusive
+			for i := 0; i < 5; i++ {
+				testlog.Infof("%d Create deployment %s with 2 Guaranteed pods requesting 2 cpus", i, DeploymentName)
+				dp = deployments.Make(DeploymentName, testutils.NamespaceTesting,
+					deployments.WithPodTemplate(p),
+					deployments.WithNodeSelector(testutils.NodeSelectorLabels))
+				dp.Spec.Template.Annotations = annotations
+				dp.Spec.Template.Labels = deploymentLabels // Add labels to the pod template
+				dp.Spec.Replicas = &numberofReplicas
+
+				Expect(testclient.DataPlaneClient.Create(ctx, dp)).ToNot(HaveOccurred())
+				podList := &corev1.PodList{}
+				listOptions := &client.ListOptions{Namespace: testutils.NamespaceTesting, LabelSelector: labels.SelectorFromSet(deploymentLabels)}
+				Eventually(func() bool {
+					if err := testclient.DataPlaneClient.List(context.TODO(), podList, listOptions); err != nil {
+						return false
+					}
+					for _, pod := range podList.Items {
+						Expect(pod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+						if isPodReady(&pod) || pod.Status.Phase == corev1.PodRunning {
+							continue
+						}
+						for _, containerStatus := range pod.Status.ContainerStatuses {
+							if containerStatus.State.Waiting == nil {
+								continue
+							}
+							// we want to test container Pending state first so that we do not skip the condition
+							// since we are creating and deleting the deployments, it's possible for this condition
+							// to be skipped if one of the pods is not in pending state
+							if containerStatus.State.Waiting.Reason == "RunContainerError" && strings.Contains(containerStatus.State.Waiting.Message, "failed to run pre-start hook for container") {
+								testlog.Infof("container %s failed to start with error: %s", pod.Spec.Containers[0].Name, containerStatus.State.Waiting.Message)
```

Comment:
nit: I'd go with Warningf instead of Infof to be aligned with the rest of the reports in the eventually when it returns false